File: data/multilang_comparison/program_cpp_vector/src/func20.c
#include "program_cpp_vector.h" 
array_t* func20(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array168;
      if (pCounter > 0) {
         array168 = vars->data[--pCounter];
         array168->refC++;
         DEBUG_COPY(array168->id);
      } else {
         array168 = (array_t*)malloc(sizeof(array_t));
         array168->size = 631;
         array168->refC = 1;
         array168->id = 168;
         array168->data = (unsigned int*)malloc(array168->size*sizeof(unsigned int));
         memset(array168->data, 0, array168->size*sizeof(unsigned int));
         DEBUG_NEW(array168->id);
      }
      unsigned int loop50 = 0;
      unsigned int loopLimit50 = (rand()%loopsFactor)/3 + 1;
      for(; loop50 < loopLimit50; loop50++) {
         for (int i = 0; i < array168->size; i++) {
            array168->data[i]--;
         }
         for (int i = 0; i < array168->size; i++) {
            if (array168->data[i] == 27) { 
               return array168;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array168;
      array_t* array169 = func28(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array169->id);
      free(params0.data);
      array169->refC--;
      if(array169->refC == 0) {
         free(array169->data);
         free(array169);
         DEBUG_FREE(array169->id);
      }
      array168->refC--;
      if(array168->refC == 0) {
         free(array168->data);
         free(array168);
         DEBUG_FREE(array168->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array170 = func22(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array170->id);
      free(params0.data);
      array170->refC--;
      if(array170->refC == 0) {
         free(array170->data);
         free(array170);
         DEBUG_FREE(array170->id);
      }
   }
   array_t* array171;
   if (pCounter > 0) {
      array171 = vars->data[--pCounter];
      array171->refC++;
      DEBUG_COPY(array171->id);
   } else {
      array171 = (array_t*)malloc(sizeof(array_t));
      array171->size = 512;
      array171->refC = 1;
      array171->id = 171;
      array171->data = (unsigned int*)malloc(array171->size*sizeof(unsigned int));
      memset(array171->data, 0, array171->size*sizeof(unsigned int));
      DEBUG_NEW(array171->id);
   }
   return array171;
}

File: data/multilang_comparison/program_go/program_go/src/func20.go
package main
import "math/rand"
var _ = rand.Intn
func Func20(vars *ArrayParam, PATH0 uint64 , loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   if (PATH0 & 1) != 0 {
      var array168 *Array
      if pCounter > 0 {
          array168 = vars.Data[pCounter-1]
          pCounter--
          array168.RefC++
      } else {
          array168 = &Array{
              Size: 631,
              RefC: 1,
              Id: 168,
              Data: make([]uint32, 631),
          }
      }
      var loop50 int = 0;
      var loopLimit50 int = (50)/3 + 1;
      for ; loop50 < loopLimit50; loop50++ {
         for i := 0; i < array168.Size; i++ {
             array168.Data[i]--
         }
         for i := 0; i < array168.Size; i++ {
             if array168.Data[i] == 27 {
                 return array168
             }
         }
      }
      var params0 ArrayParam
      params0.Size = 1
      params0.Data = make([]*Array, params0.Size)
      params0.Data[0] = array168
      var array169 *Array = Func28(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array169.RefC--
      if array169.RefC == 0 {
          array169.Data = nil
          array169 = nil
      }
      array168.RefC--
      if array168.RefC == 0 {
          array168.Data = nil
          array168 = nil
      }
   
   } else {
      var params0 ArrayParam
      params0.Size = 0
      params0.Data = make([]*Array, params0.Size)
      var array170 *Array = Func22(&params0, getPath(), loopsFactor);
      params0.Data = nil
      params0.Size = 0
      array170.RefC--
      if array170.RefC == 0 {
          array170.Data = nil
          array170 = nil
      }
   }
   var array171 *Array
   if pCounter > 0 {
       array171 = vars.Data[pCounter-1]
       pCounter--
       array171.RefC++
   } else {
       array171 = &Array{
           Size: 512,
           RefC: 1,
           Id: 171,
           Data: make([]uint32, 512),
       }
   }
   return array171;
}

